src/main/java/br/ufpe/cin/files/FilesManager.java

```diff
@@ -220,17 +230,77 @@ private FilesManager(){}
 	 * @return string content of the file, or null in case of errors.
 	 */
 	public static String readFileContent(File file){
+		
 		//StringBuilder content = new StringBuilder();
 		String content = "";
 		try{
-			BufferedReader reader = Files.newBufferedReader(Paths.get(file.getAbsolutePath()), StandardCharsets.UTF_8);
+			String fileEncoding = encodings.get(file);
```

Comment:
Que tal chamar o `detectEncoding` diretamente aqui?

Reply:
@gaabs mais pra frente eu vou precisar fazer uma mudança no parser envolvendo enconding, então fica mais útil chamando lá no ```mergeFiles```

Reply:
Foi só por uma questão de performance. Esse método é chamado várias vezes durante um merge, mas o encoding de um arquivo não muda. Então ao invés de chamar o algoritmo que detecta o encoding sempre que for ler, eu chamo apenas uma vez antes de realizar o merge, e esse método só vê qual é.